### PR TITLE
[Modules] New module Azure Digital Twins

### PR DESCRIPTION
**Pull request by AlexanderSehr**
Follow-up PR to #2135 

---

Original description by @ryanmstephens:

Created new module for Azure Digital Twins: https://azure.microsoft.com/en-gb/products/digital-twins/#overview

New module doesn't have any dependencies. This module creates Azure Digital Twins Instances and Digital Twins Instances Endpoints. To the following services: Event Hub, Event Grid and Service Bus.

## Pipeline references

| Pipeline |
| - |
| [Build Status](https://dev.azure.com/carml/CARML/_build/latest?definitionId=135&branchName=users%2Falsehr%2F2135_digTw) |

# Type of Change

- [x] New feature (non-breaking change which adds functionality)

# Checklist

- [x] I'm sure there are no other open Pull Requests for the same update/change
- [x] My corresponding pipelines / checks run clean and green without any errors or warnings
- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation (readme)
- [x] I did format my code

### CODE REVIEW

File: modules/Microsoft.DigitalTwins/digitalTwinsInstances/readme.md [hunk 1]
```diff
@@ -0,0 +1,649 @@
...
+
+<details>
+
+<summary>Bicep format</summary>
+
+```bicep
+  EndpointName: '<Endpoint Name>'
+  EventHubEntityPath: 'evh1' // Event Hub Name
+  EventHubURI: 'sb://xyz.servicebus.windows.net' //Event Hub namespace, including sb://
+```
```
Comment (MariusStorhaug) [inline modules/Microsoft.DigitalTwins/digitalTwinsInstances/readme.md:321]:
```suggestion
```bicep
  eventhubEndpoint: {
    EndpointName: '<Endpoint Name>'
    EventHubEntityPath: 'evh1' // Event Hub Name
    EventHubURI: 'sb://xyz.servicebus.windows.net' //Event Hub namespace, including sb://
  }
```
```

File: modules/Microsoft.DigitalTwins/digitalTwinsInstances/readme.md [hunk 2]
```diff
@@ -0,0 +1,649 @@
...
+
+<details>
+
+<summary>Bicep format</summary>
+
+```bicep
+   EndpointName: '<Endpoint Name>'
+   EventHubConnectionStringPrimaryKey: 'Endpoint=sb://xyz.servicebus.windows.net/;SharedAccessKeyName=<SAS-Key-Name>;SharedAccessKey=<SAS-KEY>=;EntityPath=<Event-Hub Name>'
+   EventHubConnectionStringSecondaryKey: 'Endpoint=sb://xyz.servicebus.windows.net/;SharedAccessKeyName=<SAS-Key-Name>;SharedAccessKey=<SAS-Key>;EntityPath=<Event-Hub Name>'
+```
```
Comment (MariusStorhaug) [inline modules/Microsoft.DigitalTwins/digitalTwinsInstances/readme.md:353]:
```suggestion
```bicep
  eventhubEndpoint: {
    EndpointName: '<Endpoint Name>'
    EventHubConnectionStringPrimaryKey: 'Endpoint=sb://xyz.servicebus.windows.net/;SharedAccessKeyName=<SAS-Key-Name>;SharedAccessKey=<SAS-KEY>=;EntityPath=<Event-Hub Name>'
    EventHubConnectionStringSecondaryKey: 'Endpoint=sb://xyz.servicebus.windows.net/;SharedAccessKeyName=<SAS-Key-Name>;SharedAccessKey=<SAS-Key>;EntityPath=<Event-Hub Name>'
  }
```
```

File: modules/Microsoft.DigitalTwins/digitalTwinsInstances/readme.md [hunk 3]
```diff
@@ -0,0 +1,649 @@
...
+<details>
+
+<summary>Bicep format</summary>
+
+```bicep
+  EndpointName: '<Endpoint Name>'
+  EventGridTopicEndpoint: 'https://xyz.<region>.eventgrid.azure.net/api/events'
+  EventGridAccessKey1: '<Event Grid Topic Access Key 1>'
+  EventGridAccessKey2: '<Event Grid Topic Access Key 2>'
+```
```
Comment (MariusStorhaug) [inline modules/Microsoft.DigitalTwins/digitalTwinsInstances/readme.md:386]:
```suggestion
```bicep
  eventgridEndpoint: {
    EndpointName: '<Endpoint Name>'
    EventGridTopicEndpoint: 'https://xyz.<region>.eventgrid.azure.net/api/events'
    EventGridAccessKey1: '<Event Grid Topic Access Key 1>'
    EventGridAccessKey2: '<Event Grid Topic Access Key 2>'
  }
```
```

File: modules/Microsoft.DigitalTwins/digitalTwinsInstances/readme.md [hunk 4]
```diff
@@ -0,0 +1,649 @@
...
+
+<details>
+
+<summary>Bicep format</summary>
+
+```bicep
+    EndpointName: '<Endpoint Name>'
+    ServiceBusEndpointUri: 'sb://xyz.servicebus.windows.net' //Service Bus URI, including sb://
+    ServiceBusEntityPath: '<Service Bus Entity Path>'
+```
```
Comment (MariusStorhaug) [inline modules/Microsoft.DigitalTwins/digitalTwinsInstances/readme.md:419]:
```suggestion
```bicep
  servicebusEndpoint: {
    EndpointName: '<Endpoint Name>'
    ServiceBusEndpointUri: 'sb://xyz.servicebus.windows.net' //Service Bus URI, including sb://
    ServiceBusEntityPath: '<Service Bus Entity Path>'
  }
```
```

File: modules/Microsoft.DigitalTwins/digitalTwinsInstances/readme.md [hunk 5]
```diff
@@ -0,0 +1,649 @@
...
+
+<details>
+
+<summary>Bicep format</summary>
+
+```bicep
+    EndpointName: '<Endpoint Name>'
+    ServiceBusPrimaryConnectionString: 'Endpoint=sb://xyz.servicebus.windows.net/;SharedAccessKeyName=<SAS Key Name>;SharedAccessKey=<SAS Key>=;EntityPath=<EntityPath>'
+    ServiceBusSecondaryConnectionString: 'Endpoint=sb://xyz.servicebus.windows.net/;SharedAccessKeyName=<SAS Key Name>;SharedAccessKey=<SAS Key>=;EntityPath=<EntityPath>'
+```
```
Comment (MariusStorhaug) [inline modules/Microsoft.DigitalTwins/digitalTwinsInstances/readme.md:452]:
```suggestion
```bicep
  servicebusEndpoint: {
    EndpointName: '<Endpoint Name>'
    ServiceBusPrimaryConnectionString: 'Endpoint=sb://xyz.servicebus.windows.net/;SharedAccessKeyName=<SAS Key Name>;SharedAccessKey=<SAS Key>=;EntityPath=<EntityPath>'
    ServiceBusSecondaryConnectionString: 'Endpoint=sb://xyz.servicebus.windows.net/;SharedAccessKeyName=<SAS Key Name>;SharedAccessKey=<SAS Key>=;EntityPath=<EntityPath>'
  }
```
```

File: modules/Microsoft.DigitalTwins/digitalTwinsInstances/readme.md [hunk 6]
```diff
@@ -0,0 +1,649 @@
...
+Deployment of Digital Twins Event Hub Endpoint. Identity based authentication.
+
+<details>
+
+<summary>Parameter JSON format</summary>
+
+```json
+        "eventhubEndpoint": {
+            "value": {
+                "EndpointName": "<Endpoint Name>",
```
Comment (MariusStorhaug) [inline modules/Microsoft.DigitalTwins/digitalTwinsInstances/readme.md:304]: Is it intentional to use pascal casing here on properties of the object?

**Comment by AlexanderSehr** [comment]
Closed in favor of #3039

**Pull request closed without merging**